applications/tari_console_wallet/src/grpc/wallet_grpc_server.rs

```diff
@@ -656,6 +658,42 @@ impl wallet_server::Wallet for WalletGrpcServer {
         Ok(Response::new(ImportUtxosResponse { tx_ids }))
     }
 
+    async fn submit_contract_acceptance(
+        &self,
+        request: Request<SubmitContractAcceptanceRequest>,
+    ) -> Result<Response<SubmitContractAcceptanceResponse>, Status> {
+        let mut asset_manager = self.wallet.asset_manager.clone();
+        let mut transaction_service = self.wallet.transaction_service.clone();
+        let message = request.into_inner();
+
+        let contract_id = FixedHash::try_from(message.contract_id)
+            .map_err(|err| Status::invalid_argument(format!("Invalid contract_id:{:?}", err)))?;
+
+        let validator_node_public_key =
+            PublicKey::from_bytes(message.validator_node_public_key.as_slice()).map_err(|e| {
+                Status::invalid_argument(format!("Validator node public key was not a valid pub key:{}", e))
+            })?;
+
+        let signature = Signature::try_from(message.signature.unwrap())
```

Comment:
Not providing a signature will cause a panic

```suggestion
let signature = message.signature.ok_or_else(|| Status::bad_request("signature not provided"))?;
        let signature = Signature::try_from(signature)
```